### PR TITLE
Basic MPN standardization

### PR DESCRIPTION
**Pull request by subbaraonv**

### CODE REVIEW

File: util-core/src/main/scala/com/indix/utils/core/MPN.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.indix.utils.core
+
+import org.apache.commons.lang3.StringUtils
+import org.apache.commons.lang3.text.WordUtils
+
+object MPN {
+  // Some domain specific keywords known to be invalid
+  val BlackListedMpns = Set("unknown", "none", "does not apply", "non applicable", "various")
```
Comment (codingnirvana) [inline util-core/src/main/scala/com/indix/utils/core/MPN.scala]: Lets also add n/a and na to this list

File: util-core/src/main/scala/com/indix/utils/core/MPN.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  // Does not consider numbers or one word strings as title-case phrase
+  def isTitleCase(str: String): Boolean = {
+    val words = str.split(' ').filter(_.nonEmpty)
+    if (words.length < 2) false
+    else words.forall(w => w == WordUtils.capitalizeFully(w) && !StringUtils.isNumeric(w))
+  }
+
+  def standardizeMPN(input: String): String = {
```
Comment (codingnirvana) [inline util-core/src/main/scala/com/indix/utils/core/MPN.scala]: Why not use the return type as Option[String]?

Comment (codingnirvana) [reply util-core/src/main/scala/com/indix/utils/core/MPN.scala]: While we add this as an UDF, that part of the code can have the return type as string.

File: util-core/src/test/scala/com/indix/utils/core/MPNTest.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    MPN.isTitleCase("Samsung Galaxy A8") should be(true)
+    MPN.isTitleCase("Tempered Glass") should be(true)
+
+    MPN.isTitleCase("1442820G1") should be(false)
+    MPN.isTitleCase("Macbook") should be(false)
+    MPN.isTitleCase("CE 7200") should be(false)
+  }
+
+  it should "validate identifier" in {
+    MPN.isValidIdentifier("") should be (false)
```
Comment (codingnirvana) [inline util-core/src/test/scala/com/indix/utils/core/MPNTest.scala]: lets also add the `null` test case

**Comment by subbaraonv** [comment]
@codingnirvana please check now

**Comment by codingnirvana** [comment]
LGTM. Merging it.

**Pull request merged**